SourceCode/GPS/Forms/Profiles/FormNewProfile.cs

```diff
@@ -81,42 +81,52 @@ private void textBoxName_TextChanged(object sender, EventArgs e)
         private void buttonOK_Click(object sender, EventArgs e)
         {
             string newProfileName = SanitizeFileName(textBoxName.Text.Trim()).Trim();
-            if (!string.IsNullOrEmpty(newProfileName))
+            if (string.IsNullOrEmpty(newProfileName))
+                return;
+
+            string newProfilePath = Path.Combine(RegistrySettings.vehiclesDirectory, newProfileName + ".xml");
+
+            // Bestaat al? -> bevestigen om te overschrijven
+            if (File.Exists(newProfilePath))
             {
-                string newProfilePath = Path.Combine(RegistrySettings.vehiclesDirectory, newProfileName + ".xml");
+                var overwrite = FormDialog.Show(
+                    gStr.gsSaveAndReturn,
+                    $"Profile '{newProfileName}' already exists.\r\n\r\nOverwrite?",
+                    MessageBoxButtons.YesNo);
 
-                if (File.Exists(newProfilePath))
-                {
-                    DialogResult result = FormDialog.Show(
-                        gStr.gsSaveAndReturn,
-                        $"Profile '{newProfileName}' already exists. Overwrite?",
-                        MessageBoxButtons.YesNo);
-
-                    if (result != DialogResult.OK)
-                    {
-                        return;
-                    }
-                }
+                if (overwrite != DialogResult.Yes)
+                    return;
```

Comment:
Shouldn't that be `DialogResult.OK`? Otherwise I think it will always return here.
```suggestion
                if (overwrite != DialogResult.OK)
                    return;
```